src/Voron/StorageEnvironmentOptions.cs

```diff
@@ -250,6 +292,79 @@ public override IJournalWriter CreateJournalWriter(long journalNumber, long jour
                 return result.Value;
             }
 
+            private string GetJournalForReuse(long journalNumber)
+            {
+                var newJournalName = JournalName(journalNumber);
+                var path = Path.Combine(_journalPath, newJournalName);
+
+                if (File.Exists(path))
+                    return null; // journal might be already exists from previous session
+
+                Tuple<string, DateTime> reused = null;
+
+                var retries = 3;
+                while (retries-- > 0)
+                {
+                    if (_journalsForReuse.TryPop(out reused) == false)
+                        return null;
+
+                    try
+                    {
+                        File.Move(reused.Item1, path);
+                        break;
+                    }
+                    catch (Exception ex)
+                    {
+                        if (_log.IsInfoEnabled)
+                            _log.Info(
+                                "Failed to rename " + reused.Item1 + " to " + path + ". Will retry another " + retries +
+                                " times and then create regular new journal #" +
+                                journalNumber, ex);
+                        if (retries < 1)
+                            return null;
+                    }
+                }
+
+                if (reused == null)
+                {
+                    if (_log.IsInfoEnabled)
```

Comment:
No it isn't.  We TryPop from _journalsForReuse and if it is empty (at start i.e.) then we returning null from this method